### PR TITLE
1762780: fix bug, when reporter_id is empty; ENT-1706

### PR DESCRIPTION
**Pull request by jirihnidek**
* BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1762780
* Validation using _validate_non_empty_string did not have side
  effect of setting default value and debug print also didn't
  print default value

### CODE REVIEW

**Comment by jirihnidek** [comment]
It needs more testing, because I changed validation method.

File: virtwho/config.py [hunk 1]
```diff
@@ -825,9 +825,18 @@ def _validate_non_empty_string(self, key):
                 result = ('warning', 'Value for %s not set' % key)
         else:
             if not isinstance(value, str):
-                result = ('warning', '%s is not set to a valid string, using default' % key)
+                if self.has_default(key):
+                    self._values[key] = self.defaults[key]
+                    result = ('warning', '%s is not set to a valid string, using default: %s' %
+                              (key, self.defaults[key]))
+                else:
+                    result = ('warning', '%s is not set to a valid string and has default value: %s' % key)
```
Comment (wottop) [inline virtwho/config.py]: has *no* default value?

**Review by wottop — CHANGES_REQUESTED** [review]

File: virtwho/config.py (hunk 1 above)
Comment (jirihnidek) [reply virtwho/config.py]: Good point. Thanks.

**Comment by wottop** [comment]
ack

**Pull request merged**